File: source/code/dsa/linked_lists/python/reverse_iterative.py
def reverse_linked_list(head):
    """Reverse the items in a linked list and return the new head"""
    current_node = head
    previous_node = None
    while current_node is not None:
        next_node = current_node.next
        current_node.next = previous_node
        previous_node = current_node
        current_node = next_node
    return previous_node

File: source/code/dsa/linked_lists/javascript/reverse_iterative.js
/*
    * Reverses a linked list and returns the new head.
    */
function reverseLinkedList(head) {
    let previous = null;
    let current = head;
    while (current !== null) {
        let next = head.next;
        current.next = previous;
        previous = current;
        current = next;
    }
    return previous;
}
